src/libs/dutil/butil.cpp

```diff
@@ -214,15 +212,12 @@ HRESULT DAPI BundleEnumRelatedBundle(
 
         if (fUpgradeCodeFound)
         {
-            hr = RegReadString(hkBundle, BUNDLE_REGISTRATION_REGISTRY_BUNDLE_PROVIDER_KEY, &szProviderKey );
-            ExitOnFailure(hr, "Failed to read the bundle provider key.");
-
-            hr = ::StringCchLengthW(szProviderKey, STRSAFE_MAX_CCH, reinterpret_cast<UINT_PTR*>(&cchProviderKey));
-            ExitOnFailure(hr, "Failed to calculate length of string");
-
             if (lpBundleIdBuf)
             {
-                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, szProviderKey, cchProviderKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
+                hr = ::StringCchLengthW(sczUninstallSubKey, STRSAFE_MAX_CCH, reinterpret_cast<UINT_PTR*>(&cchUninstallSubKey));
+                ExitOnFailure(hr, "Failed to calculate length of string");
+
+                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, sczUninstallSubKey, cchUninstallSubKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
```

Comment:
StringCchCatNExW is going to fail after the first GUID if the limit is MAX_GUID_CHARS. Any reason not to use StringCchCopy*?